src/nunavut/lang/docgen/__init__.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Copyright (C) 2018-2019  UAVCAN Development Team  <uavcan.org>
+# This software is distributed under the terms of the MIT License.
+#
+"""
+    Filters for generating docs. All filters in this
+    module will be available in the template's global namespace as ``docgen``.
+"""
+
+import string
+import random
+import typing
+
+import pydsdl
+
+from ...templates import template_language_filter
+from .. import Language
+
+
+@template_language_filter(__name__)
+def filter_is_composite(language: Language, instance: typing.Any) -> bool:
+    return isinstance(instance, pydsdl.CompositeType)
+
+
+@template_language_filter(__name__)
+def filter_is_array(language: Language, instance: typing.Any) -> bool:
+    return isinstance(instance, pydsdl.ArrayType)
+
+
+@template_language_filter(__name__)
+def filter_is_service(language: Language, instance: typing.Any) -> bool:
+    return isinstance(instance, pydsdl.ServiceType)
+
+
+@template_language_filter(__name__)
+def filter_is_service_req(language: Language, instance: typing.Any) -> bool:
+    return instance.has_parent_service and instance.full_name.split(".")[-1] == "Request"
+
+
+@template_language_filter(__name__)
+def filter_is_field(language: Language, instance: typing.Any) -> bool:
+    return isinstance(instance, pydsdl.Field)
```

Comment:
If you rebase to main you can replace these filters with tests (available since #200):

https://github.com/UAVCAN/nunavut/blob/c3c56efec1626c667e087695144882c23642c79a/src/nunavut/lang/__init__.py#L82-L84